File: python/enolib/elements/embed.py
from . import section
from .value_element_base import ValueElementBase


class Embed(ValueElementBase):
    def __repr__(self):
        return f"<class Embed key={self._instruction['key']} value={self._print_value()}>"

    def parent(self):
        return self._parent or section.Section(self._context, self._instruction['parent'])

File: ruby/lib/enolib/elements/embed.rb
# frozen_string_literal: true

module Enolib
  class Embed < ValueElementBase
    def parent
      @parent || Section.new(@context, @instruction[:parent])
    end

    def to_s
      "#<Enolib::Embed key=#{@instruction[:key]} value=#{print_value}>"
    end
  end
end
